superset/db_engine_specs/hive.py

```diff
@@ -429,12 +429,26 @@ def execute(  # type: ignore
 
     @classmethod
     @cache.memoize()
```

Comment:
Somewhat unrelated to your change I noticed there a relative new caching structure defined [here](https://github.com/apache/incubator-superset/blob/master/superset/utils/cache.py#L27) which provides caching at the database level. I'm not sure if there's merit in using that in favor of the global cache, but I do wonder if we should be using the `tables_cache` rather than the `cache` for table, i.e., SQL Lab, related caching.

Note it's all a little confusing when a cached object expires given the plethora of caches which now seem to exist.

Reply:
Thanks for pointing this out, I completely agree with this assessment.